bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/scheduling/Scheduler.java

```diff
@@ -234,25 +240,18 @@ public Runner(int id) {
 
         @Override
         public void run() {
-            runSafely(id);
-        }
-    }
-
-    /**
-     * Defensive programming to prevent unforeseen errors.
-     */
-    private void runSafely(int id) {
-        try {
-            run(id);
-        } catch (Throwable t) {
-            RuntimeUtils.printCrashLog(t);
+            try {
+                runTask(id);
+            } catch (Throwable t) {
+                RuntimeUtils.printCrashLog(t);
```

Comment:
Can the runTask fail with an error?

Reply:
Most probably no. This is defensive programming. If we did not handle this, the scheduler may hang. We have done this earlier as well. I just moved it.

Reply:
ack